apps/ops/migrations/0005_historicaljob_periodic_variable_and_more.py

```diff
@@ -0,0 +1,53 @@
+        ('ops', '0004_job_nodes_alter_job_assets'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='historicaljob',
+            name='periodic_variable',
+            field=models.JSONField(default=dict, verbose_name='Periodic variable'),
+        ),
+        migrations.AddField(
+            model_name='job',
+            name='periodic_variable',
+            field=models.JSONField(default=dict, verbose_name='Periodic variable'),
+        ),
+        migrations.CreateModel(
+            name='Variable',
+            fields=[
+                ('created_by', models.CharField(blank=True, max_length=128, null=True, verbose_name='Created by')),
+                ('updated_by', models.CharField(blank=True, max_length=128, null=True, verbose_name='Updated by')),
+                ('date_created', models.DateTimeField(auto_now_add=True, null=True, verbose_name='Date created')),
+                ('date_updated', models.DateTimeField(auto_now=True, verbose_name='Date updated')),
+                ('comment', models.TextField(blank=True, default='', verbose_name='Comment')),
+                ('id', models.UUIDField(default=uuid.uuid4, primary_key=True, serialize=False)),
+                ('name', models.CharField(max_length=1024, null=True, verbose_name='Name')),
+                ('var_name', models.CharField(help_text="The variable name used in the script has a fixed prefix 'jms_' followed by the input variable name. For example, if the variable name is 'name,' the final generated environment variable will be 'jms_name'.", max_length=1024, null=True, verbose_name='Variable name')),
+                ('default_value', models.CharField(max_length=2048, null=True, verbose_name='Default Value')),
+                ('type', models.CharField(default='text', max_length=64, verbose_name='Variable type')),
+                ('tips', models.CharField(blank=True, default='', max_length=1024, null=True, verbose_name='Tips')),
+                ('required', models.BooleanField(default=False, verbose_name='Required')),
+                ('extra_args', models.JSONField(default=dict, verbose_name='ExtraVars')),
+                ('adhoc', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='variable', to='ops.adhoc', verbose_name='Adhoc')),
+                ('creator', models.ForeignKey(null=True, on_delete=django.db.models.deletion.SET_NULL, to=settings.AUTH_USER_MODEL, verbose_name='Creator')),
+                ('job', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='variable', to='ops.job', verbose_name='Job')),
+                ('playbook', models.ForeignKey(null=True, on_delete=django.db.models.deletion.CASCADE, related_name='variable', to='ops.playbook', verbose_name='Playbook')),
+            ],
+            options={
+                'verbose_name': 'Variable',
+                'ordering': ['date_created'],
+            },
+        ),
+    ]
```

Comment:
There don't appear to be any issues with this migration database schema. It appears to have been completed successfully according to the "Generated by Django" timestamp. You may want to review and approve it manually if working directly within a development environment.